dot/parachain/types/types.go

```diff
@@ -323,6 +349,25 @@ type CandidateReceipt struct {
 	CommitmentsHash common.Hash `scale:"2"`
 }
 
+// Encode returns the SCALE encoding of the receipt.
+func (cr CandidateReceipt) Encode() ([]byte, error) {
```

Comment:
Do we need custom encode methods here?

Reply:
I believe this is being used by the backend while storing the data. It does make the code simple if we abstract it out. Is there a reason not to use custom encode method for certain types?

Reply:
There are some cases where custom encoding is required. But unless it is actually required (data is not scale encoded, to meet an interface constraint, etc) there probably is not need.

For example, unless there is context that I am missing, this Encode function is just a wrapper around marshalling the data. IMO this is not necessary because on the caller side it is the exact same amount of complexity, and to me it makes more sense to just marshal directly rather than calling a separate encode method that simply just calls Marshal.

Reply:
I believe this is being used to hash the `CandidateReceipt` and we need to scale encode it to get the hash. 

Since this is being used only in the `Hash()` method of `CandidateReceipt`, I will remove it for now and maybe create the `Encode()` method if it is being used in multiple places in the future.

Thank you!